packages/contracts-bedrock/src/safe/DeputyPauseModule.sol

```diff
@@ -0,0 +1,221 @@
+
+    /// @notice Getter function for the maximum signature validity in seconds.
+    /// @return maxSignatureValiditySeconds_ Maximum signature validity in seconds.
+    function maxSignatureValiditySeconds() public view returns (uint256 maxSignatureValiditySeconds_) {
+        maxSignatureValiditySeconds_ = MAX_SIGNATURE_VALIDITY_SECONDS;
+    }
+
+    /// @notice Getter function for the Pause message typehash.
+    /// @return pauseMessageTypehash_ Pause message typehash.
+    function pauseMessageTypehash() public pure returns (bytes32 pauseMessageTypehash_) {
+        pauseMessageTypehash_ = PAUSE_MESSAGE_TYPEHASH;
+    }
+
+    /// @notice Getter function for the DeputyAuth message typehash.
+    /// @return deputyAuthMessageTypehash_ DeputyAuth message typehash.
+    function deputyAuthMessageTypehash() public pure returns (bytes32 deputyAuthMessageTypehash_) {
+        deputyAuthMessageTypehash_ = DEPUTY_AUTH_MESSAGE_TYPEHASH;
+    }
+
+    /// @notice Calls the Foundation Safe's `execTransactionFromModuleReturnData()` function with
+    ///         the arguments necessary to call `pause()` on the Security Council Safe, which will
+    ///         then cause the Security Council Safe to trigger SuperchainConfig pause.
+    ///         Front-running this function is completely safe, it'll pause either way.
+    /// @param _expiry Signature expiry timestamp.
+    /// @param _nonce Signature nonce.
+    /// @param _signature ECDSA signature.
+    function pause(uint256 _expiry, bytes32 _nonce, bytes memory _signature) external {
+        // Check that the signature is not expired.
+        if (_expiry <= block.timestamp) {
+            revert DeputyPauseModule_SignatureExpired();
+        }
+
+        // Check that the signature expiry is not too far in the future.
+        if (_expiry - block.timestamp > MAX_SIGNATURE_VALIDITY_SECONDS) {
+            revert DeputyPauseModule_SignatureExpiryTooLong();
+        }
+
+        // Make sure the nonce hasn't been used yet.
+        if (usedNonces[_nonce]) {
+            revert DeputyPauseModule_NonceAlreadyUsed();
+        }
+
+        // Verify the signature.
+        bytes32 digest = _hashTypedData(keccak256(abi.encode(PAUSE_MESSAGE_TYPEHASH, PauseMessage(_expiry, _nonce))));
+        if (ECDSA.recover(digest, _signature) != DEPUTY) {
+            revert DeputyPauseModule_Unauthorized();
+        }
+
+        // Mark the nonce as used.
+        usedNonces[_nonce] = true;
+
+        // Attempt to trigger the call.
+        (bool success, bytes memory returnData) = FOUNDATION_SAFE.execTransactionFromModuleReturnData(
+            address(DEPUTY_GUARDIAN_MODULE), 0, abi.encodeCall(IDeputyGuardianModule.pause, ()), Enum.Operation.Call
+        );
+
+        // If the call fails, revert.
+        if (!success) {
+            revert DeputyPauseModule_ExecutionFailed(string(returnData));
+        }
+
+        // Emit the Paused event and note who triggered it.
+        emit Paused("Pause Deputy");
```

Comment:
The superchain pause itself already emits an event, and we can always trace a transaction to find the originator of a pause, so curious what value this adds?

Reply:
Yeah, agree here, I was just following the pattern that was already in place in the `DeputyGuardianModule`.

Reply:
I think @maurelian and I agreed at some point that the string identifier never really added value here, so I think it's ok to remove the event from here